content/blog/gargle-1-2-0/index.Rmd

```diff
@@ -0,0 +1,243 @@
+```
+
+The dance concludes with gargle (or a wrapper package) in possession of a [Google user token](https://developers.google.com/identity/protocols/oauth2#installed).
+This credential usually comes with an access token, which is valid for about an hour, and a more persistent refresh token, which can be used to get new access tokens in the future.
+It is nice to store this refresh token somewhere, to reduce future friction for this user.
+We'll refer to this "somewhere" as the *oauth token cache*.
+
+One of the original motivations for the gargle package was to depart from httr's behaviour around token caching, which defaults to the current working directory.[^1]
+By default, gargle caches user tokens centrally, at the user level, and their keys or labels also convey which Google identity is associated with each token.
+This reduces the chance of accidentally sending tokens to remote servers (like DropBox or GitHub), increases the ability to reuse tokens across multiple R projects, and makes it easier to use multiple identities within a single project.
+
+[^1]: The successor to httr is under development and it will also cache user tokens at the user level, like gargle.
+
+Various incremental improvements have been made around gargle's token cache:
+
+* The default cache location has moved, to better align with general conventions
+  around where to cache user data. After upgrading gargle, don't be surprised
+  if you see some messages about cache relocation.
+* Token storage relies on serialized R objects and therefore is potentially
+  sensitive to differences in R version and operating system between when a
+  token is created and when it is (re)used. gargle is now able to recognize and
+  cope with the most predictable and harmless effects of moving a token across
+  time or space.
+* gargle is quite conservative about using cached tokens and likes to have
+  explicit permission from the user. However, this is in tension with the desire
+  to make things "just work". Balancing these two goals is an ongoing effort:
+  - In a non-interactive context, gargle will use a cached token, if it can
+    find (at least) one, even if the user has not given explicit instructions.
+    However, we complain noisily and urge the user to make their intent
+    unambiguous in the code. This comes up, for example, when rendering R
+    Markdown documents.
+  - You can now identify the token you want to use with a domain-only glob-like
+    pattern. This makes it possible to successfully execute code like this,
+    non-interactively, on the machines of both `alice@example.com` and
+    `bob@example.com`.
+    
+    ```{r, eval = FALSE}
+    library(googledrive)
+    
+    # googledrive uses gargle for auth
+    drive_auth(email = "*@example.com")
+    ```
+
+## Rolling the built-in OAuth client
+
+User tokens created with gargle are the result of so-called "three-legged OAuth", where the three legs are the user, Google, and an OAuth client (id and secret).
```

Comment:
"3-legged" is an outdated term; I'd suggest re-using "OAuth dance" from above

Reply:
Reworded.